homeassistant/components/rituals_perfume_genie/__init__.py

```diff
@@ -30,27 +31,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     try:
         account_devices = await account.get_devices()
-    except ClientConnectorError as ex:
-        raise ConfigEntryNotReady from ex
+    except aiohttp.ClientError as err:
+        raise ConfigEntryNotReady from err
 
-    hublots = []
     devices = {}
     for device in account_devices:
         hublot = device.data[HUB][HUBLOT]
-        hublots.append(hublot)
         devices[hublot] = device
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
         COORDINATORS: {},
         DEVICES: devices,
     }
 
-    for hublot in hublots:
-        device = hass.data[DOMAIN][entry.entry_id][DEVICES][hublot]
+    for hublot, device in devices.items():
 
         async def async_update_data():
-            await device.update_data()
-            return device.data
+            try:
+                await device.update_data()
+            except aiohttp.ClientError as err:
```

Comment:
ClientError is already caught by the coordinator.

https://github.com/home-assistant/core/blob/6604614c399610cc40341e99da278627499b35a1/homeassistant/helpers/update_coordinator.py#L188-L193